api/src/test/java/bio/terra/tanagra/service/query/api/VariableScopeTest.java

```diff
@@ -0,0 +1,36 @@
+package bio.terra.tanagra.service.query.api;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import bio.terra.tanagra.service.search.Entity;
+import bio.terra.tanagra.service.search.EntityVariable;
+import bio.terra.tanagra.service.search.Variable;
+import java.util.Optional;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+@Tag("unit")
+public class VariableScopeTest {
+  private static final Entity FOO = Entity.builder().underlay("u").name("foo").build();
+  private static final Entity BAR = Entity.builder().underlay("u").name("bar").build();
+
+  private static final EntityVariable X_FOO = EntityVariable.create(FOO, Variable.create("x"));
+  private static final EntityVariable Y_FOO = EntityVariable.create(FOO, Variable.create("y"));
+  private static final EntityVariable Z_FOO = EntityVariable.create(FOO, Variable.create("z"));
+
+  @Test
+  void scope() {
```

Comment:
May be worth covering a variable shadowing case here to at least fix the semantics. I'm not sure whether or not it should be disallowed.

Reply:
Thanks, yep, added a couple tests.